docs/source/concepts/local-ledger.rst

```diff
@@ -0,0 +1,296 @@
+   This section focuses on a single DAML ledger and therefore abstracts from these events.
+   The :ref:`Ledger Interoperability <ledger-interoperability>` section extends the concepts in this section accordingly.
+
+The definition uses the following concepts:
+
+.. _def-action-activation-deactivation:
+
+Definition »activation, use, deactivation, allocation, deallocation«
+  Every action allocates, uses, or deactivates a contract or it allocates, uses, or deallocates a key.
+
+  * A **Create** action of a contract **activates the contract**.
+    If the created contract has a key, the action also **allocates the key** to the created contract.
+
+  * An **Exercise** action on a contract **uses the contract**.
+    If the exericse is consuming, it also **deactivates the contract** and, if the contract has a key, it additionally **deallocates the key**.
+
+  * A **Fetch** action on a contract **uses the contract**.
+
+  * A **NoSuchKey** action on a key **uses the key**.
+
+Definition »action order«
+  For a directed acyclic graph of transactions,
+  the induced **action order** on the actions in the transactions combines graph-induced order between transactions with the total order of actions inside every transaction.
+  It is the least partial order that includes the following ordering relations between two actions `act`:sub:`1` and `act`:sub:`2`:
+  
+  * `act`:sub:`1` and `act`:sub:`2` belong to the same transaction and `act`:sub:`1` precedes `act`:sub:`2` in the transaction.
+  * `act`:sub:`1` and `act`:sub:`2` belong to different transactions in vertices `tx`:sub:`1` and `tx`:sub:`2` and there is a path from `tx`:sub:`1` to `tx`:sub:`2`.
+
+The action order is a partial order on the actions in a directed acyclic graph of transactions.
+Moreover, whenever `act`:sub:`1` is action-ordered before another action `act`:sub:`2` in a directed acyclic graph of transactions,
+then the graph contains a (possibly empty) path from `act`:sub:`1`\ 's transaction to `act`:sub:`2`\ 's transaction.
+
+Definition »local ledger«
+  A **local ledger** for a party `A` is a finite directed acyclic graph that satisfies the following conditions:
+
+  * *Total order on contract changes:*
+    For every contract that `A` is a stakeholder of, the action order totally orders all activations and deactivations of the contract.
+    In this sequence, activations alternate with deactivations, starting with an activation.
+
+  * *Total order on key changes:*
+    For every key that `A` is a maintainer of, the action order totally orders all allocations and deallocations of the key.
+    In this sequence, allocations alternate with deallocations, starting with an allocation, and allocation-deallocations pairs refer to the same contract.
+
+  * *Use ordering:*
+    If `A` is a stakeholder of a contract and an informee of an action `act`:sub:`u` that uses the contract,
```

Comment:
No. In the last PR, we agreed that we do not want to totally order all uses of a contract. Accordingly, I cannot require a total order on all "known uses" of a contract, because the additional ordering between uses would induce additional ordering between transactions and thus strengthen causality. Corda does not provide any ordering guarantees between different uses of a contract.

I don't exactly know what you mean with "known usage". If `A` is an observer, it may witness a use action of a contract (e.g., a fetch) even though it is not an informee of the use action. In that case, this witnessed use must not induce an order between actions or transactions for `A`.

Reply:
Oh, indeed, my bad.